editors/sc-ide/core/sc_server.cpp

```diff
@@ -314,56 +319,66 @@ void ScServer::sendVolume( float volume )
 }
 
 bool ScServer::isRecording() const { return mIsRecording; }
+bool ScServer::isPaused() const { return mIsRecordingPaused; }
 
-void ScServer::setRecording( bool doRecord )
-{
-    static const QString startRecordingCommand("ScIDE.defaultServer.record");
-    static const QString stopRecordingCommand("ScIDE.defaultServer.stopRecording");
 
-    if (!isRunning() || mIsRecording == doRecord)
-        return;
 
-    mIsRecording = doRecord;
+void ScServer::setRecording( bool doRecord )
+    {
+        if (!isRunning())
+            return;
 
-    if (doRecord) {
-        mLang->evaluateCode( startRecordingCommand );
-        mRecordTime = system_clock::now();
-        mRecordTimer.start();
-    }
-    else {
-        mLang->evaluateCode( stopRecordingCommand );
-        mRecordTimer.stop();
-    }
+        mIsRecording = doRecord;
+        mIsRecordingPaused = false;
+        updateRecordingAction();
+}
 
-    updateRecordingAction();
+void ScServer::pauseRecording( bool flag )
+    {
+        if(mIsRecordingPaused != flag) {
+            mIsRecordingPaused = flag;
+            if(flag) {
+                mLang->evaluateCode( QString("ScIDE.defaultServer.pauseRecording"), true );
+            } else {
+                if(mIsRecording) {
+                setRecording(true);
+                sendRecording(true);
+               }
+            }
+            updateRecordingAction();
+        }
 }
 
-seconds ScServer::recordingTime() const
+
+void ScServer::sendRecording( bool doRecord )
 {
-    if (isRecording())
-        return duration_cast<seconds>( system_clock::now() - mRecordTime );
-    else
-        return seconds(0);
+    static const QString startRecordingCommand("ScIDE.defaultServer.record");
+    static const QString stopRecordingCommand("ScIDE.defaultServer.stopRecording");
```

Comment:
`static const QString` is so qt4 ... we have qt5 and `QStringLiteral`, which is more efficient